app/src/main/java/org/schabi/newpipe/util/external_communication/TextLinkifier.java

```diff
@@ -0,0 +1,281 @@
+     *
+     * @param context          the context to use
+     * @param markdownBlock    the block of markdown text to be linked
+     * @param textView         the TextView to set the plain text block linked
+     * @param streamingService the {@link StreamingService} of the content
+     * @param contentUrl       the URL of the content
+     * @return a disposable to be stored somewhere and disposed when activity/fragment is destroyed
+     */
+    public static Disposable createLinksFromMarkdownText(final Context context,
+                                                         final String markdownBlock,
+                                                         final TextView textView,
+                                                         final StreamingService streamingService,
+                                                         final String contentUrl) {
+        final Markwon markwon = Markwon.builder(context).usePlugin(LinkifyPlugin.create()).build();
+        markwon.setMarkdown(textView, markdownBlock);
+        return changeIntentsOfDescriptionLinks(context, textView.getText(), textView,
+                streamingService, contentUrl);
+    }
+
+    /**
+     * Add click listeners which opens a search on hashtags in a plain text.
+     * <p>
+     * This method finds all timestamps in the {@link SpannableStringBuilder} of the description
+     * using a regular expression, adds for each a {@link ClickableSpan} which opens
+     * {@link NavigationHelper#openSearch(Context, int, String)} and makes a search on the hashtag,
+     * in the service of the content.
+     *
+     * @param context              the context to use
+     * @param spannableDescription the SpannableStringBuilder with the text of the
+     *                             content description
+     * @param streamingService     the {@link StreamingService} of the content
+     */
+    private static void addClickListenersOnHashtags(final Context context,
+                                                    final SpannableStringBuilder
+                                                            spannableDescription,
+                                                    final StreamingService streamingService) {
+        final String descriptionText = spannableDescription.toString();
+        final Matcher hashtagsMatches = HASHTAGS_PATTERN.matcher(descriptionText);
+
+        while (hashtagsMatches.find()) {
+            final int hashtagStart = hashtagsMatches.start(1);
+            final int hashtagEnd = hashtagsMatches.end(1);
+            final String parsedHashtag = descriptionText.substring(hashtagStart, hashtagEnd);
+
+            spannableDescription.setSpan(new ClickableSpan() {
+                @Override
+                public void onClick(@NonNull final View view) {
+                    NavigationHelper.openSearch(context, streamingService.getServiceId(),
+                            parsedHashtag);
```

Comment:
Shouldn't you search for the value of the hashtag without the `#` prefixed? (maybe not)

Reply:
No, I forgot this part.

Reply:
Should be done now.